examples/simpleGov/README.md

```diff
@@ -0,0 +1,1064 @@
+    AbstainVotes    int64           // Total number of Abstain votes
+}
+```
+
+In terms of store, we will just create one KVStore in the multistore to store `Proposals`. We will also store the vote (`Yes`, `No` or `Abstain`) chosen by each voter on each proposal.
+
+- `Proposals` will be indexed by `'proposals'|<proposalID>`.
+- `Votes` (`Yes`, `No`, `Abstain`) will be indexed by `'proposals'|<proposalID>|'votes'|<voterAddress>`.
+
+Notice the quote mark on `'proposals'` and `'votes'`. They indicate that these are constant keywords. So, for example, the option casted by voter with address `0x01` on proposal `0101` will be stored at index `'proposals'|0101|'votes'|0x01`.
+
+These keywords are used to faciliate range queries. Range queries (TODO: Link to formal spec) allow developer to query a subspace of the store, and return an iterator. They are made possible by the nice properties of the [IAVL+ tree](https://github.com/tendermint/iavl) that is used in the background. In practice, this means that it is possible to store and query a Key-Value pair in O(1), while still being able to iterate over a given subspace of Key-Value pairs. For example, we can query all the addresses that voted on a given proposal, along with their votes, by calling `rangeQuery(SimpleGovStore, <proposalID|'addresses'>)`.
+
+### Transactions
+
+The title of this section is a bit misleading. Indeed, what you, as a module developer, have to define are not `Transactions`, but `Messages`. Both transactions and messages exist in the Cosmos-SDK, but a transaction differs from a message in that a message is contained in a transaction. Transactions wrap around messages and add standard information like signatures and fees. As a module developer, you do not have to worry about transactions, only messages.
+
+Let us define the messages we need in order to modify the state. Based on our features, we only need two messages:
+
+- `SubmitProposalMsg`: to submit proposals
+- `VoteMsg`: to vote on proposals
+
+```go
+type SubmitProposalMsg struct {
+    Title           string      // Title of the proposal
+    Description     string      // Description of the proposal
+    Deposit         sdk.Coins   // Deposit paid by submitter. Must be > MinDeposit to enter voting period
+    Submitter       sdk.Address // Address of the submitter
+}
+```
+
+```go
+type VoteMsg struct {
+    ProposalID  int64           // ID of the proposal
+    Option      string          // Option chosen by voter
+    Voter       sdk.Address     // Address of the voter
+}
+```
+
+## Implementation
+
+Now, that we have our types defined, we can start actually implementing the module.
+
+First, let us go into the module's folder and create a folder for our module.
+
+```bash
+cd x/
+mkdir simple_governance
+cd simple_governance
+```
+
+Let us start by adding the files we will need. Your module's folder should look something like that:
+
+```
+x
+└─── simple_governance
```

Comment:
After reading through the whole tutorial it might be helpful to link the concepts in this tutorial like (`keepers`, `handlers`) to the files in this diagram. Also not showing the `*_test.go` files in the file hierarchy might make this diagram more readable/informative.